hive-metastore/src/main/java/org/apache/iceberg/hive/HMSTablePropertyHelper.java

```diff
@@ -0,0 +1,261 @@
+            });
+    setCommonParameters(
+        newMetadataLocation,
+        metadata.uuid(),
+        obsoleteProps,
+        currentLocation,
+        parameters,
+        BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE.toUpperCase(Locale.ENGLISH),
+        metadata.schema(),
+        maxHiveTablePropertySize);
+    setStorageHandler(parameters, hiveEngineEnabled);
+
+    // Set the basic statistics
+    if (summary.get(SnapshotSummary.TOTAL_DATA_FILES_PROP) != null) {
+      parameters.put(StatsSetupConst.NUM_FILES, summary.get(SnapshotSummary.TOTAL_DATA_FILES_PROP));
+    }
+    if (summary.get(SnapshotSummary.TOTAL_RECORDS_PROP) != null) {
+      parameters.put(StatsSetupConst.ROW_COUNT, summary.get(SnapshotSummary.TOTAL_RECORDS_PROP));
+    }
+    if (summary.get(SnapshotSummary.TOTAL_FILE_SIZE_PROP) != null) {
+      parameters.put(StatsSetupConst.TOTAL_SIZE, summary.get(SnapshotSummary.TOTAL_FILE_SIZE_PROP));
+    }
+
+    setSnapshotStats(metadata, parameters, maxHiveTablePropertySize);
+    setPartitionSpec(metadata, parameters, maxHiveTablePropertySize);
+    setSortOrder(metadata, parameters, maxHiveTablePropertySize);
+
+    tbl.setParameters(parameters);
+  }
+
+  public static void setHmsTableParameters(
+      String newMetadataLocation,
+      Table tbl,
+      ViewMetadata metadata,
+      Set<String> obsoleteProps,
+      long maxHiveTablePropertySize,
+      String currentLocation) {
+    Map<String, String> parameters =
+        Optional.ofNullable(tbl.getParameters()).orElseGet(Maps::newHashMap);
+
+    // push all Iceberg view properties into HMS
+    metadata.properties().entrySet().stream()
+        .filter(entry -> !entry.getKey().equalsIgnoreCase(HiveCatalog.HMS_TABLE_OWNER))
+        .forEach(entry -> parameters.put(entry.getKey(), entry.getValue()));
+    setCommonParameters(
+        newMetadataLocation,
+        metadata.uuid(),
+        obsoleteProps,
+        currentLocation,
+        parameters,
+        HiveOperationsBase.ICEBERG_VIEW_TYPE_VALUE.toUpperCase(Locale.ENGLISH),
+        metadata.schema(),
+        maxHiveTablePropertySize);
+    tbl.setParameters(parameters);
+  }
+
+  private static void setCommonParameters(
+      String newMetadataLocation,
+      String uuid,
+      Set<String> obsoleteProps,
+      String currentLocation,
+      Map<String, String> parameters,
+      String tableType,
+      Schema schema,
+      long maxHiveTablePropertySize) {
+    if (uuid != null) {
+      parameters.put(TableProperties.UUID, uuid);
+    }
+
+    obsoleteProps.forEach(parameters::remove);
+
+    parameters.put(BaseMetastoreTableOperations.TABLE_TYPE_PROP, tableType);
+    parameters.put(BaseMetastoreTableOperations.METADATA_LOCATION_PROP, newMetadataLocation);
+
+    if (currentLocation != null && !currentLocation.isEmpty()) {
+      parameters.put(BaseMetastoreTableOperations.PREVIOUS_METADATA_LOCATION_PROP, currentLocation);
+    }
+    setSchema(schema, parameters, maxHiveTablePropertySize);
```

Comment:
Please add a new line:
```suggestion

    setSchema(schema, parameters, maxHiveTablePropertySize);
```

Reply:
Fixed.